From f3a3eeefdbca33d92643dcc313737145c5889c9b Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:04:43 -0400
Subject: [fs] btrfs: remove unused wait queue in struct extent_buffer

Message-id: <1407179109-13714-397-git-send-email-sandeen@redhat.com>
Patchwork-id: 87373
O-Subject: [RHEL7.1 PATCH 396/422] Btrfs: remove unused wait queue in struct extent_buffer
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe Manana <fdmanana@gmail.com>

commit 46fefe41b5360106ebfed228fbfba62f75ad4fcd
Author: Filipe Manana <fdmanana@gmail.com>
Date:   Mon Jun 16 13:14:25 2014 +0100

    Btrfs: remove unused wait queue in struct extent_buffer

    The lock_wq wait queue is not used anywhere, therefore just remove it.
    On a x86_64 system, this reduced sizeof(struct extent_buffer) from 320
    bytes down to 296 bytes, which means a 4Kb page can now be used for
    13 extent buffers instead of 12.

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent_io.h b/fs/btrfs/extent_io.h
index 15ce5f2..ccc264e 100644
--- a/fs/btrfs/extent_io.h
+++ b/fs/btrfs/extent_io.h
@@ -158,7 +158,6 @@ struct extent_buffer {
   * to unlock
   */
  wait_queue_head_t read_lock_wq;
- wait_queue_head_t lock_wq;
  struct page *pages[INLINE_EXTENT_BUFFER_PAGES];
 #ifdef CONFIG_BTRFS_DEBUG
  struct list_head leak_list;
-- 
1.7.1